pay-api/src/pay_api/models/payment.py

```diff
@@ -203,10 +209,33 @@ def search_purchase_history(cls,  # noqa:E501; pylint:disable=too-many-arguments
             query = query.filter(Invoice.created_name.ilike(f'%{created_name}%'))
         if invoice_id := search_filter.get('id', None):
             query = query.filter(cast(Invoice.id, String).like(f'%{invoice_id}%'))
+        if payment_type := search_filter.get('paymentMethod', None):
+            if payment_type == 'NO_FEE':
```

Comment:
Currently it only exists in the UI as an enum since its specific to what we see visually in the table. It's not something that's used in the backend, but we should think about adding it as a payment type instead of setting them to INTERNAL I think (there would probably be some repercussions there though)